lib/plugins/aws/provider.js

```diff
@@ -1592,6 +1593,26 @@ class AwsProvider {
     return `${provider.deploymentPrefix}`;
   }
 
+  initS3DirectoryPaths(provider) {
+    const date = new Date();
+    const serviceStage = `${this.serverless.service.service}/${this.getStage()}`;
+    const dateString = `${date.getTime().toString()}-${date.toISOString()}`;
+    const prefix = this.provider.getDeploymentPrefix();
+
+    provider.s3CodeArtifactsDirectoryPath = `${prefix}/${serviceStage}/code-artifacts`;
+    provider.s3DeploymentDirectoryPath = `${prefix}/${serviceStage}/${dateString}`;
```

Comment:
Let's not set it on `service.provider`, this object reflects user configuration and ideally if we do not mix it with internally resolved data (although I know that currently our internals in few places do that).

Ideally if `s3CodeArtifactsDirectoryPath` and `s3DeploymentDirectoryPath` getters simply resolve it on spot
And _timestamp_ part can be set on provider instance in its constructor as e.g. `this.deploymentTimestamp = ...`